hw02-generics/src/main/java/ru/otus/homework/CustomerService.java

```diff
@@ -0,0 +1,39 @@
+package ru.otus.homework;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    // todo: 3. надо реализовать методы этого класса
+    // важно подобрать подходящую Map-у, посмотрите на редко используемые методы, они тут полезны
+    private final TreeMap<Customer, String> customersMap = new TreeMap<>(Comparator.comparing(Customer::getScores));
```

Comment:
Тип слева лучше объявлять минимально необходимым интерфейсом.
```suggestion
    private final NavigableMap<Customer, String> customersMap = new TreeMap<>(Comparator.comparing(Customer::getScores));
```

Reply:
В эту сторону как-то не подумал, спасибо за замечание. Исправил